yapapi/strategy.py

```diff
@@ -71,18 +77,22 @@ def __str__(self):
 class MarketStrategy(DemandDecorator, abc.ABC):
     """Abstract market strategy."""
 
-    acceptable_prop_value_ranges: Dict[str, PropValueRange]
+    acceptable_prop_value_ranges_overrides: Dict[str, PropValueRange]
+    __acceptable_prop_value_ranges: Dict[str, PropValueRange]
+
+    @property
+    def acceptable_prop_value_ranges(self) -> Dict[str, PropValueRange]:
+        if not hasattr(self, "_acceptable_prop_value_ranges"):
```

Comment:
We have `acceptable_prop_value_ranges` with either two underscores or no underscores. Here you check for `acceptable_prop_value_ranges` with a single underscore, I guess this is not what you wanted?

Reply:
Also: I **still** don't think we need such weird construction.
I'd say `hasattr` and `getattr` are for special strange cases, and we're **not** doing anything special here.

Reply:
they're not for "special strange cases", they're excatly what they're for